src/test/java/org/folio/rest/api/CheckInStorageApiTest.java

```diff
@@ -0,0 +1,138 @@
+import org.folio.rest.support.http.AssertingRecordClient;
+import org.folio.rest.support.http.InterfaceUrls;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import io.vertx.core.json.JsonObject;
+
+public class CheckInStorageApiTest extends ApiTests {
+  private static final String TABLE_NAME = "check_in";
+
+  private final AssertingRecordClient checkInClient =
+    new AssertingRecordClient(client, StorageTestSuite.TENANT_ID,
+      InterfaceUrls::checkInsStorageUrl, "checkIns");
+
+  @Before
+  public void beforeEach() {
+    StorageTestSuite.cleanUpTable(TABLE_NAME);
+  }
+
+  @After
+  public void checkIdsAfterEach() {
+    StorageTestSuite.checkForMismatchedIDs(TABLE_NAME);
+  }
+
+  @Test
+  public void canCreateCheckIn() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = createSampleCheckIn().create();
+
+    IndividualResource createResult = checkInClient.create(checkInToCreate);
+
+    assertThat(createResult.getJson(), is(checkInToCreate));
+  }
+
+
+  @Test
+  public void cannotCreateCheckInIfRequiredPropertyMissing() throws InterruptedException,
+    ExecutionException, TimeoutException, MalformedURLException {
+
+    JsonObject checkInToCreate = new CheckInBuilder().create();
+
+    JsonResponse createResponse = checkInClient.attemptCreate(checkInToCreate);
+
+    assertThat(createResponse, isValidationResponseWhich(hasMessage("may not be null")));
+    assertThat(createResponse, isValidationResponseWhich(hasParameter("occurredDateTime", "null")));
+  }
+
+  @Test
+  public void canSearchForCheckInsByOccurredDateTime()
+    throws InterruptedException, ExecutionException, TimeoutException,
+    MalformedURLException {
+
+    final DateTime beforeCreateDateTime = DateTime.now(DateTimeZone.UTC);
+
+    JsonObject firstCheckIn = createSampleCheckIn().create();
+    JsonObject secondCheckIn = createSampleCheckIn().create();
+    JsonObject thirdCheckIn = createSampleCheckIn().create();
+
+    checkInClient.create(firstCheckIn);
+    checkInClient.create(secondCheckIn);
+    checkInClient.create(thirdCheckIn);
+
+    MultipleRecords<JsonObject> allCheckInOperations = checkInClient
+      .getMany(String.format("occurredDateTime >= %s", beforeCreateDateTime.getMillis()));
+
+    assertThat(allCheckInOperations.getTotalRecords(), is(3));
+    assertTrue(allCheckInOperations.getRecords().contains(firstCheckIn));
+    assertTrue(allCheckInOperations.getRecords().contains(secondCheckIn));
+    assertTrue(allCheckInOperations.getRecords().contains(thirdCheckIn));
+  }
+
+  @Test
+  public void cannotUseNegativeOffsetForSearch()
+    throws InterruptedException, ExecutionException, TimeoutException,
+    MalformedURLException {
+
+    JsonResponse jsonResponse = checkInClient
+      .attemptGetMany("occurredDateTime is not null", -10, 10);
```

Comment:
Is `is not null` a valid CQL query?

Reply:
No, the test should fail before processing CQL by RMB.

Reply:
I think it would better demonstrate the intent of this test if the CQL was valid.

That would also mean the test might not start failing if the RAML Module Builder changes the order in which it validates requests